inference-engine/src/vpu/graph_transformer/src/stages/gather_nd.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <vpu/frontend/frontend.hpp>
+
+#include <memory>
+#include <string>
+
+namespace vpu {
+
+namespace {
+
+class GatherNDStage final : public StageNode {
+public:
+    using StageNode::StageNode;
+
+protected:
+    StagePtr cloneImpl() const override { return std::make_shared<GatherNDStage>(*this); }
+
+    void propagateDataOrderImpl(StageDataInfo<DimsOrder> &orderInfo) override {
+        const auto input1 = inputEdge(0)->input();
+        const auto input2 = inputEdge(1)->input();
+        const auto output = outputEdge(0)->output();
+
+        orderInfo.setInput(inputEdge(0), DimsOrder::fromNumDims(input1->desc().numDims()));
+        orderInfo.setInput(inputEdge(1), DimsOrder::fromNumDims(input2->desc().numDims()));
+        orderInfo.setOutput(outputEdge(0), DimsOrder::fromNumDims(output->desc().numDims()));
+    }
+
+    void getDataStridesRequirementsImpl(StageDataInfo<StridesRequirement> &stridesInfo) override {
+        for (const auto &inEdge : inputEdges()) {
+            stridesInfo.setInput(inEdge, StridesRequirement::compact());
+        }
+        stridesInfo.setOutput(outputEdge(0), StridesRequirement::compact());
+    }
+
+    void finalizeDataLayoutImpl() override {}
+
+    void getBatchSupportInfoImpl(StageDataInfo<BatchSupport> &batchInfo) override {}
+
+    StageSHAVEsRequirements getSHAVEsRequirementsImpl() const override {
+        return StageSHAVEsRequirements::NotNeeded;
+    }
+
+    void initialCheckImpl() const override {
+        VPU_THROW_UNLESS(numInputs() == 2, "Number of inputs must be 2");
+        VPU_THROW_UNLESS(numOutputs() == 1, "Number of outputs must be 1");
+        VPU_THROW_UNLESS(inputs()[0]->desc().type() == outputs()[0]->desc().type(),
+                         "First input and output must have the same DataType");
+
+        assertInputsOutputsTypes(
+            this,
+            {{DataType::U8, DataType::FP16, DataType::S32}, {DataType::S32}},
+            {{DataType::U8, DataType::FP16, DataType::S32}});
+    }
+
+    void serializeParamsImpl(BlobSerializer &serializer) const override {
+        const auto batchDims = attrs().get<int32_t>("batch_dims");
```

Comment:
`getOrDefault`?

Reply:
As I understand, there always be a value, because in `FrontEnd::parseGatherND`
```
const auto batchDims = layer->GetParamAsInt("batch_dims", 0); // 0 is a default value

stage->attrs().set<int32_t>("batch_dims", batchDims);
```

Reply:
It is just about further safety. If we implement `addGatherNDStage` then a developer will have the opportunity to set attributes by oneself